### PR TITLE
[natives] expose all feature flags inside NativeContext as an extension

### PR DESCRIPTION
**Pull request by alinush**
### Description

Exposes the feature flags inside the Rust implementation of Move native functions. This is useful for the generic elliptic-curves Move module (https://github.com/aptos-labs/aptos-core/pull/6550), which will be upgradeable by feature gating newly-added curves directly inside the Rust natives.

### CODE REVIEW

**Review by zjma — APPROVED** [review]

**Review by vgao1996 — CHANGES_REQUESTED** [review]
Hmm... I feel like this may not be the best way to expose feature flags to native functions.

Extensions are rather heavy-weight and involve dynamic casting. Instead, you can pass the features to the `make_all` function in each crate, which will then pass them down to each native function. I'd also recommend you add a `features: Features` field to the `SafeNativeContext` and make it available to natives. Notice that `SafeNativeContext` already contains the timed features, so we just need to replicate that pattern for features.

**Comment by vgao1996** [comment]
I'm aware that `struct Features` contains a vector, so we may not want to clone it often. Feel free to put it in an Arc before passing it down to the natives. Alternatively you could also change its representation from Vec<u8> to something like SmallVec, but I don't know if that's gonna work with serde.

**Comment by alinush** [comment]
> I'm aware that `struct Features` contains a vector, so we may not want to clone it often. Feel free to put it in an Arc before passing it down to the natives. Alternatively you could also change its representation from Vec to something like SmallVec, but I don't know if that's gonna work with serde.

Makes sense to me! I'll use an `std::sync::Arc` then & see how things work out.

File: aptos-move/aptos-vm/src/move_vm_ext/vm.rs [hunk 1]
```diff
@@ -23,6 +24,7 @@ use std::ops::Deref;
 pub struct MoveVmExt {
     inner: MoveVM,
     chain_id: u8,
+    features: Features,
```
Comment (alinush) [inline aptos-move/aptos-vm/src/move_vm_ext/vm.rs:27]: @vgao1996 , I'll just change this to an `Arc` and pass it around. Makes sense?

**Review by vgao1996 — APPROVED** [review]

**Pull request merged**